src/hooks/membershipCardTerms.ts

```diff
@@ -222,15 +214,15 @@ export const useMembershipCardTerms = (cardId: string) => {
             const details = await (strategyMap[discount.product.type] || strategyMap['default'])({
               productId: discount.product.target,
               queryClient,
-              type: discount.product.type,
+              type: discount.product.type as MembershipCardTermsProductType,
             })
 
             return {
               id: discount.id,
               type: discount.type,
               amount: discount.amount,
               product: {
-                type: discount.product.type,
+                type: discount.product.type as MembershipCardTermsProductType,
```

Comment:
adding specified type here , i think  it wasn't what I designed it for in the beginning

here , i use strategyMap to implement OCP , that because , I don't want to worry about what type of products are available in this section of the code,

in the future , new product will update only strategyMap, and then we don't need to worry about this section of code

Reply:
OK, adjusted.